api/v1/checks.go

```diff
@@ -163,6 +163,35 @@ func (c S3BucketCheck) GetType() string {
 	return "s3Bucket"
 }
 
+type ResticCheck struct {
+	// Description about the current restic canary check
+	Description string `yaml:"description,omitempty" json:"description,omitempty"`
+	// Repository The restic repository path eg: rest:https://user:pass@host:8000/ or rest:https://host:8000/ or s3:s3.amazonaws.com/bucket_name
+	Repository string `yaml:"repository" json:"repository"`
+	// Password for the restic repository
+	Password string `yaml:"password" json:"password"`
+	// MaxAge for backup freshness
+	MaxAge string `yaml:"maxAge" json:"maxAge"`
+	// AccessKey access key id for connection with aws s3, minio, wasabi, alibaba oss
+	AccessKey string `yaml:"accessKey,omitempty" json:"accessKey,omitempty"`
+	// SecretKey secret access key for connection with aws s3, minio, wasabi, alibaba oss
+	SecretKey string `yaml:"secretKey,omitempty" json:"secretKey,omitempty"`
+	// CaCert path to the root cert. In case of self-signed certificates
+	CaCert string `yaml:"caCert,omitempty" json:"caCert,omitempty"`
```

Comment:
This should probably be InsecureTLSSkipVerify like in http checlk

Reply:
Okay... though in restic skipping a [TLS/Verification is not possible](https://github.com/restic/restic/issues/2656) user needs to define the path of CaRoot cert. I took this field as same in helmCheck